routes/campgrounds.js

```diff
@@ -5,18 +5,41 @@ const middleware = require('../middleware');
 
 // INDEX - show all campgrounds in the web
 router.get('/', (req, res) => {
-  Campground.find({}, (err, allCampgrounds) => {
-    if (err) {
-      console.log(err);
-    } else {
-      res.render('campgrounds/index', { campgrounds: allCampgrounds });
-    }
-  });
+  let noMatch = null;
+  if (req.query.search) {
+    const regex = new RegExp(escapeRegex(req.query.search), 'gi');
```

Comment:
The name `regex` is misleading. I'd suggest using `escapedSearchQuery` or something similar.